lib/ansible/modules/cloud/amazon/ec2_asg.py

```diff
@@ -1035,25 +1102,47 @@ def create_autoscaling_group(connection):
             max_size = as_group['MaxSize']
         if desired_capacity is None:
             desired_capacity = as_group['DesiredCapacity']
-        launch_config_name = launch_config_name or as_group['LaunchConfigurationName']
-
-        try:
-            launch_configs = describe_launch_configurations(connection, launch_config_name)
-        except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
-            module.fail_json(msg="Failed to describe launch configurations",
-                             exception=traceback.format_exc())
-        if len(launch_configs['LaunchConfigurations']) == 0:
-            module.fail_json(msg="No launch config found with name %s" % launch_config_name)
         ag = dict(
             AutoScalingGroupName=group_name,
-            LaunchConfigurationName=launch_configs['LaunchConfigurations'][0]['LaunchConfigurationName'],
             MinSize=min_size,
             MaxSize=max_size,
             DesiredCapacity=desired_capacity,
             HealthCheckGracePeriod=health_check_period,
             HealthCheckType=health_check_type,
             DefaultCooldown=default_cooldown,
             TerminationPolicies=termination_policies)
+        # Validate the Launch Config / Launch Template provided exists.
+        if launch_config_name:
+            try:
+                launch_configs = describe_launch_configurations(connection, launch_config_name)
+            except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+                module.fail_json(msg="Failed to describe launch configurations",
+                                 exception=traceback.format_exc())
+            if len(launch_configs['LaunchConfigurations']) == 0:
+                module.fail_json(msg="No launch config found with name %s" % launch_config_name)
+            ag['LaunchConfigurationName'] = launch_configs['LaunchConfigurations'][0]['LaunchConfigurationName']
+        elif launch_template:
+            try:
+                lt = describe_launch_templates(ec2_connection, launch_template)['LaunchTemplates'][0]
+            except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+                module.fail_json(msg="Failed to describe launch templates",
+                                 exception=traceback.format_exc())
+            if not lt:
+                module.fail_json(msg="No launch template found matching %s" % launch_template,
+                                 exception=traceback.format_exc())
+            # Prefer LaunchTemplateId over LaunchTemplateName as it's more specific.
+            ag['LaunchTemplate'] = {"LaunchTemplateId": lt['LaunchTemplateId'], "Version": str(launch_template['version'])}
```

Comment:
Use suboptions you can force the options to be strings rather than needing to cast it here.

Oh - and now I see you have used suboptions, nice! So you can remove str()